### DISCUSSION TITLE
multi-schema client

### DISCUSSION THREAD

**Discussion by milesje** [supabase/supabase#12344] [Questions] [answered]
Is it possible to configure the client to allow access to multiple schemas? Or to create multiple client instances for each schema? I've tried to create a second client passing the the options `{db: {schema: 'my-other-schema'}}` but I get the following error.

`Object { code: "PGRST106", details: null, hint: null, message: "The schema must be one of the following: public, storage, graphql_public" }`

Is there something I need to do to the schemas via the dashboard or using an SQL command?

**Comment by GaryAustin1** [comment]
The 2nd method.  But you need to use dashboard API settings to add the schema and your schema needs to have grants set for anon,authenticated, etc roles.

**Comment by milesje** [comment, marked as answer]
So this is my first time using Supabase and  I'm working on a SvelteKit project. I was having issues getting a second client instance working in the +page.js files load function. So I moved the functionality to the +page.svelte file's onMount function and it is now working. I'm not 100% sure why it wasn't working correctly in the laod function as that is where the logic really needs to be. I will continue to work on getting it in the load function.
